### PR TITLE
if the folder doesn't exist

### PR DESCRIPTION
**Pull request by AminTorabi-NOAA**
In the current implementation, if the user specifies a directory for `stream_output_directory` (e.g., `output/`) that doesn't exist, it results in an error. With the new PR, the code automatically creates the specified directory if it doesn't exist, allowing the output to be saved in that folder without any issues.

## Additions

-

## Removals

-

## Changes

-

## Testing

1.

## Notes

-

## Todos

-

## Checklist

- [ ] PR has an informative and human-readable title
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code can be automatically merged (no conflicts)
- [ ] Code follows project standards (link if applicable)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Target Environment support

- [ ] Windows
- [ ] Linux
- [ ] Browser

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

File: src/troute-config/troute/config/output_parameters.py [hunk 1]
```diff
@@ -93,11 +93,19 @@ class ParityCheckCompareFileSet(BaseModel):
...
 class StreamOutput(BaseModel):
     # NOTE: required if writing StreamOutput files
-    stream_output_directory: Optional[DirectoryPath] = None
+    stream_output_directory: Optional[str] = None
     mask_output: Optional[FilePath] = None
     stream_output_time: int = 1
     stream_output_type:streamOutput_allowedTypes = ".nc"
     stream_output_internal_frequency: Annotated[int, Field(strict=True, ge=5)] = 5
+    
+    @validator('stream_output_directory', pre=True, always=True)
```
Comment (aaraney) [inline src/troute-config/troute/config/output_parameters.py:102]: A few minor improvements:

- We only need to validate after `pydantic` has coerced the input into a `pathlib.Path` (drop `pre=True` and `always=True`)
- Handle if we got an unexpanded path (e.g. `~/output/dir`)
- Raise if we got a file
- Create dir if it doesn't exist, and all intermediate directories

```python
    @validator('stream_output_directory')
    def validate_stream_output_directory(cls, value):
        if value is None:
            return None
            
        # expand ~/output/dir -> /home/user/output/dir
        value = value.expanduser()
        
        if value.exists() and not value.is_dir():
            raise ValueError(f"'stream_output_directory'={value!s} is a file, expected directory.")

        # make directory (and intermediates) if they don't exist
        value.mkdir(parents=True, exist_ok=True)
        return value
```

File: src/troute-config/troute/config/output_parameters.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 from pathlib import Path
 from pydantic import BaseModel, Field, validator
-
+import os
```
Comment (aaraney) [inline src/troute-config/troute/config/output_parameters.py:3]: This can be dropped.

**Review by aaraney — CHANGES_REQUESTED** [review]
Thanks for writing this up, @AminTorabi-NOAA! I left a few change requests that cover a few possible edge cases. Once these have been added, we should be good to go! Thanks again!

**Comment by AminTorabi-NOAA** [comment]
I addressed all. Thanks for the comments @aaraney

**Review by aaraney — APPROVED** [review]
Looks great! Thanks @AminTorabi-NOAA! 🎉

**Pull request merged**